gateway/api_definition.go

```diff
@@ -618,14 +613,40 @@ func (a APIDefinitionLoader) processRPCDefinitions(apiCollection string, gw *Gat
 
 			def.Proxy.ListenPath = newListenPath
 		}
-
-		spec := a.MakeSpec(&def, nil)
-		specs = append(specs, spec)
 	}
+	specs := a.prepareSpecs(apiDefs)
 
 	return specs, nil
 }
 
+func (a APIDefinitionLoader) prepareSpecs(apiDefs []nestedApiDefinition) []*APISpec {
+	var specs []*APISpec
+
+	var wg sync.WaitGroup
+	var specsMutex sync.RWMutex
+	for _, def := range apiDefs {
+		if def.CustomMiddlewareBundle != "" {
+			wg.Add(1)
```

Comment:
@lonelycode do you think we need this or just handle all the apis in the same way? The reason to do it in this way is to avoid the gateway to spawn a lot of goroutines, and instead only create an async job when the api to be loaded need to pull a bundle